chevah/keycert/ssh.py

```diff
@@ -629,6 +629,13 @@ def _fromString_BLOB(cls, blob):
         @raises BadKeyError: if the key type (the first string) is unknown.
         """
         keyType, rest = common.getNS(blob)
+
+        try:
+            # keyType is supposed to contain only ascii characters
+            keyType.decode('ascii')
+        except UnicodeDecodeError:
+            raise BadKeyError('Invalid non-ascii blob type.')
```

Comment:
I got confused by reading the RFC https://tools.ietf.org/html/rfc4251#section-5:

"Strings are also used to store text.  In that case, US-ASCII is used for internal names, and ISO-10646 UTF-8 for text that might be displayed to the user." 

But you are right, strings can also contain binary data according to that RFC.